pkg/log/log.go

```diff
@@ -213,25 +220,68 @@ func Hint(format string, args ...interface{}) {
 
 // Fail prints a message with the error symbol first, and the text in red
 func Fail(format string, args ...interface{}) {
-	log.writer.Fail(format, args...)
+	msg := fmt.Sprintf(format, args...)
+	if log.isRedactMode {
+		msg = redactMessage(msg)
+	}
+	log.writer.Fail(msg)
 }
 
 // Println writes a line with colors
 func Println(args ...interface{}) {
-	log.writer.Println(args...)
+	msg := fmt.Sprint(args...)
+	if log.isRedactMode {
+		msg = redactMessage(msg)
+	}
+	log.writer.Println(msg)
 }
 
 // Print writes a line with colors
 func Print(args ...interface{}) {
-	log.writer.Print(args...)
+	msg := fmt.Sprint(args...)
+	if log.isRedactMode {
+		msg = redactMessage(msg)
+	}
+	log.writer.Print(msg)
 }
 
 // Printf writes a line with format
 func Printf(format string, args ...interface{}) {
-	log.writer.Printf(format, args...)
+	msg := fmt.Sprintf(format, args...)
+	if log.isRedactMode {
+		msg = redactMessage(msg)
+	}
+	log.writer.Print(msg)
 }
 
 //IsInteractive checks if the writer is interactive
 func IsInteractive() bool {
 	return log.writer.IsInteractive()
 }
+
+//AddBanned adds a new
+func AddBanned(word string) {
+	log.bannedWords = append(log.bannedWords, word)
+}
+
+//StartRedact starts redacting all variables
+func StartRedact() {
+	log.isRedactMode = true
+	sort.Slice(log.bannedWords, func(i, j int) bool {
+		return len(log.bannedWords[i]) > len(log.bannedWords[j])
+	})
+}
+
+//StopRedact will show secrets and vars
+func StopRedact() {
+	log.isRedactMode = false
+}
+
+func redactMessage(message string) string {
+	s := message
+	for _, str := range log.bannedWords {
+		s = strings.ReplaceAll(s, str, "***")
```

Comment:
`***` doesn't point which original value was replaced. WDYT about replacing with the variable name such as `$FOO`?

Reply:
That could be "dangerous" if somehow a expandEnv is executed on the string and the variable is converted to the real value. I would use `*` as it is commonly used in circleci or github actions for example

Reply:
How can they expand the variable if we are just replacing the value with the key and printing it to Stdout without doing any variable expansion. I think that this could be awesome to make it clearer what's being executed. 
Also this change may require some changes on the installer if we finally do it

Reply:
Not sure if there is any possible way of doing it, but it could happen for example that we include it in the future for some reason without realizing on this. One important thing from a security perspective is to avoid to give clues to the attackers. If you provide information about variable names, you are giving them some environment variables that contain sensitive information

Also, we have to bear in mind that this output will be stored in a configmap.

Anyways, I'm not an expert on security, from a user perspective I think it would help, but in my experience, security sometimes affects usability